system-addon/content-src/lib/short-url.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * shortURL - Creates a short version of a link's url, used for display purposes
+ *            e.g. {url: http://www.foosite.com, eTLD: "com"}  =>  "foosite"
+ *
+ * @param  {obj} link A link object, which has a .url and .eTLD property, and
+ *                    optionally a .hostname property
+ * @return {string}   A short url
+ */
```

Comment:
Somehow, I think we want to make it clear what exactly the eTLD option needs to contain.   In particular, it has to be fed from something that knows about the entire public suffix list, such as nsIEffectiveTLDService, and the eTLD passed in presumably must match the hostname itself.  Perhaps an @note with an example of how this would be used if it wasn't being fed directly by a magic query?